src/modules/45-dashboards/components/AidaGenerating/__tests__/AidaGenerating.test.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react'
+import { waitFor, render, RenderResult, screen } from '@testing-library/react'
+import { TestWrapper } from '@common/utils/testUtils'
+import { useDashboardsContext } from '@dashboards/pages/DashboardsContext'
+import { Message, MessageType, VisualizationType } from '@dashboards/types/AidaTypes.types'
+import * as sharedService from 'services/custom-dashboards'
+
+import AidaGenerating from '..'
+
+const renderComponent = (messages: Message[]): RenderResult =>
+  render(
+    <TestWrapper>
+      <AidaGenerating messages={messages} />
+    </TestWrapper>
+  )
+
+const generateMockAiTile = (
+  mockDashboard: sharedService.DashboardModel
+): Promise<sharedService.GetDashboardResponse> => {
+  return new Promise(resolve => {
+    resolve({ resource: mockDashboard })
+  })
+}
```

Comment:
Just FYI, `Promise` includes a handy `Promise.resolve` function that cuts out all the fluff.
```suggestion
const generateMockAiTile = (
  mockDashboard: sharedService.DashboardModel
): Promise<sharedService.GetDashboardResponse> => {
  return Promise.resolve({ resource: mockDashboard })
}
```

Or you can just mark it as `async` and just return
```suggestion
const generateMockAiTile = async (
  mockDashboard: sharedService.DashboardModel
): Promise<sharedService.GetDashboardResponse> => {
  return { resource: mockDashboard }
}
```